Error: No latest version defined for gcc on win11-arm64

This is an error that occurred today. Thank you for taking a look. 

```
Run fortran-lang/setup-fortran@main
Run cd $(echo "/$GITHUB_ACTION_PATH" | sed -e 's/\\/\//g' -e 's/://')
++ compiler=gcc
+++ uname -s
+++ tr '[:upper:]' '[:lower:]'
++ platform=mingw64_nt-10.0-26200-arm64
++ [[ Windows == \m\a\c\O\S ]]
++ source ./setup-fortran.sh
+++ set -ex
+++++ dirname ./setup-fortran.sh
++++ cd .
++++ pwd
+++ SCRIPT_DIR=/C/a/_actions/fortran-lang/setup-fortran/main
+++ source /C/a/_actions/fortran-lang/setup-fortran/main/.github/compat/latest-versions.sh
++++ LATEST_aocc_ubuntu_24_04=5.1.0
++++ LATEST_gcc_macos_14=15
++++ LATEST_gcc_macos_15=15
++++ LATEST_gcc_macos_15_intel=15
++++ LATEST_gcc_macos_26=15
++++ LATEST_gcc_macos_26_intel=15
++++ LATEST_gcc_ubuntu_22_04=15
++++ LATEST_gcc_ubuntu_24_04=15
++++ LATEST_gcc_windows_2022=15
++++ LATEST_gcc_windows_2025=15
++++ LATEST_intel_macos_15_intel=2021.2
++++ LATEST_intel_macos_26_intel=2021.2
++++ LATEST_intel_ubuntu_22_04=2025.2
++++ LATEST_intel_ubuntu_24_04=2025.2
++++ LATEST_intel_windows_2022=2025.2
++++ LATEST_intel_windows_2025=2025.2
++++ LATEST_intel_classic_macos_14=2021.10
++++ LATEST_intel_classic_macos_15=2021.10
++++ LATEST_intel_classic_macos_15_intel=2021.10
++++ LATEST_intel_classic_macos_26=2021.10
++++ LATEST_intel_classic_macos_26_intel=2021.10
++++ LATEST_intel_classic_ubuntu_22_04=2021.12
++++ LATEST_intel_classic_ubuntu_24_04=2021.12
++++ LATEST_intel_classic_windows_2022=2021.12
++++ LATEST_intel_classic_windows_2025=2021.12
++++ LATEST_lfortran_macos_14=0.58.0
++++ LATEST_lfortran_macos_15=0.58.0
++++ LATEST_lfortran_macos_15_intel=0.58.0
++++ LATEST_lfortran_macos_26=0.58.0
++++ LATEST_lfortran_macos_26_intel=0.58.0
++++ LATEST_lfortran_ubuntu_22_04=0.58.0
++++ LATEST_lfortran_ubuntu_24_04=0.58.0
++++ LATEST_lfortran_windows_2022=0.58.0
++++ LATEST_lfortran_windows_2025=0.58.0
++++ LATEST_nvidia_hpc_ubuntu_22_04=26.1
++++ LATEST_nvidia_hpc_ubuntu_24_04=26.1
++ case $compiler in
++ version=latest
++ install_gcc mingw64_nt-10.0-26200-arm64
++ local platform=mingw64_nt-10.0-26200-arm64
++ case $platform in
++ install_gcc_choco
++ local resolved_version=latest
++ [[ latest == \l\a\t\e\s\t ]]
+++ resolve_latest_version gcc mingw
+++ local compiler=gcc
+++ local platform=mingw
++++ detect_runner_os mingw
++++ '[' -n win11-arm64 ']'
++++ case "$ImageOS" in
++++ echo win11-arm64
++++ return
+++ local runner_os=win11-arm64
++++ echo gcc
++++ tr .- __
+++ local compiler_safe=gcc
++++ echo win11-arm64
++++ tr .- __
+++ local os_safe=win11_arm64
+++ local var_name=LATEST_gcc_win11_arm64
+++ local version=
+++ '[' -z '' ']'
+++ echo 'Error: No latest version defined for gcc on win11-arm64'
Error: No latest version defined for gcc on win11-arm64
+++ return 1
++ resolved_version=
Error: Process completed with exit code 1.
```

--- a/.github/compat/matrix.yml
+++ b/.github/compat/matrix.yml
@@ -5,6 +5,7 @@ os:
   - macos-26
   - windows-2025
   - windows-2022
+  - windows-11-arm
 toolchain:
   - {compiler: gcc, version: 15}
   - {compiler: gcc, version: 14}
@@ -69,6 +70,16 @@ exclude:
     toolchain: {compiler: nvidia-hpc}
   - os: windows-2022
     toolchain: {compiler: nvidia-hpc}
+  - os: windows-11-arm
+    toolchain: {compiler: nvidia-hpc}
+  # intel/intel-classic not available on windows-11-arm
+  - os: windows-11-arm
+    toolchain: {compiler: intel}
+  - os: windows-11-arm
+    toolchain: {compiler: intel-classic}
+  # lfortran not available on windows-11-arm (no conda/miniconda on that runner)
+  - os: windows-11-arm
+    toolchain: {compiler: lfortran}
   # aocc supported only on Ubuntu 24.04
   - os: ubuntu-22.04
     toolchain: {compiler: aocc}
@@ -79,6 +90,8 @@ exclude:
   - os: windows-2025
     toolchain: {compiler: aocc}
   - os: windows-2022
+    toolchain: {compiler: aocc}
+  - os: windows-11-arm
     toolchain: {compiler: aocc}
 include:
   # macos-14 (ARM): gcc, intel-classic full range, lfortran

--- a/action.yml
+++ b/action.yml
@@ -138,6 +138,14 @@ runs:
       shell: bash
       run: mv /usr/bin/link $RUNNER_TEMP/link
 
+    - name: Get architecture (Windows)
+      id: msvc-arch
+      if: runner.os == 'Windows' && contains(inputs.compiler, 'lfortran')
+      shell: bash
+      run: echo "arch=${RUNNER_ARCH,,}" >> $GITHUB_OUTPUT
+
     - name: Setup MSVC toolchain (Windows)
       if: runner.os == 'Windows' && contains(inputs.compiler, 'lfortran')
       uses: ilammy/msvc-dev-cmd@v1.13.0
+      with:
+        arch: ${{ steps.msvc-arch.outputs.arch }}

--- a/main.sh
+++ b/main.sh
@@ -11,6 +11,8 @@ fi
 
 source ./setup-fortran.sh
 
+check_emulation "$compiler" "$platform"
+
 case $compiler in
   gcc)
     version=${VERSION:-latest}

--- a/setup-fortran.sh
+++ b/setup-fortran.sh
@@ -19,6 +19,21 @@ sudo_wrapper() {
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 source "$SCRIPT_DIR/.github/compat/latest-versions.sh"
 
+# Fail if compiler+platform would only be available via emulation.
+# Add/remove new entries here as needed.
+check_emulation()
+{
+  local compiler=$1
+  local platform=$2
+
+  case "$compiler:$platform" in
+    gcc:mingw*arm64*|gcc:mingw*aarch64*|gcc:msys*arm64*|gcc:cygwin*arm64*)
+      echo "Error: Native GCC is not available on ARM64 Windows."
+      exit 1
+      ;;
+  esac
+}
+
 # Detect the current runner OS
 detect_runner_os()
 {
@@ -52,6 +67,9 @@ detect_runner_os()
         ;;
       win25)
         echo "windows-2025"
+        ;;
+      win11-arm64)
+        echo "windows-11-arm"
         ;;
       *)
         # If not recognized, return as-is
@@ -207,6 +225,7 @@ install_gcc_apt()
 
 install_gcc_choco()
 {
+  local platform=${1:-}
   local resolved_version=$version
   if [[ "$version" == "latest" ]]; then
     resolved_version=$(resolve_latest_version "gcc" "mingw")
@@ -325,13 +344,13 @@ install_gcc()
       install_gcc_brew
       ;;
     mingw*)
-      install_gcc_choco
+      install_gcc_choco "$platform"
       ;;
     msys*)
-      install_gcc_choco
+      install_gcc_choco "$platform"
       ;;
     cygwin*)
-      install_gcc_choco
+      install_gcc_choco "$platform"
       ;;
     *)
       echo "Unsupported platform: $platform"

